framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -381,42 +381,59 @@ public void processClassTree(ClassTree classTree) {
                 validateTypeOf(im);
             }
         }
+        checkExtendsImplements(classTree);
 
+        super.visitClass(classTree, null);
+    }
+
+    /**
+     * If "@B class Y extends @A X {}", then enforce that @B must be a subtype of @A.
+     *
+     * @param classTree class tree to check
+     */
+    protected void checkExtendsImplements(ClassTree classTree) {
+        if (classTree.getSimpleName().contentEquals("")) {
```

Comment:
I thought there was a helper for this... I only found https://github.com/typetools/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java#L1206 which checks for a flag on the constructor. Maybe extracting this in a helper would be simpler.

Reply:
Oh, I also found https://github.com/typetools/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java#L138